### PR TITLE
RTMP Publish, RTC Play, Add GOP Cache

### PR DESCRIPTION
**Pull request by xiaofeilong**
当使用RTMP推流，RTC播放时，增加RTC的gop cache，减少RTC播放的起播延迟。

### CODE REVIEW

File: trunk/src/app/srs_app_config.cpp [hunk 1]
```diff
@@ -4838,6 +4838,42 @@ bool SrsConfig::get_rtc_enabled(string vhost)
...
+    if (!conf || conf->arg0().empty()) {
+        return DEFAULT;
+    }
+
+    return SRS_CONF_PERFER_FALSE(conf->arg0());
+}
+
+int SrsConfig::get_rtc_gop_cache_max_packets(std::string vhost)
+{
+    static bool DEFAULT = false;
```
Comment (duiniuluantanqin) [inline trunk/src/app/srs_app_config.cpp]: This DEFAULT still needs to be of type int.

`TRANS_BY_GPT3`

Comment (xiaofeilong) [reply trunk/src/app/srs_app_config.cpp]: It should be used int. Just finished modifying.

`TRANS_BY_GPT3`

**Comment by duiniuluantanqin** [comment]
This mechanism seems to conflict with the low latency of RTC.

`TRANS_BY_GPT3`

**Comment by xiaofeilong** [comment]
> This mechanism seems to conflict with the low latency of RTC.

This depends on the specific application scenario. For large-scale low-latency live streaming with one-to-many connections, increasing the GOP (Group of Pictures) can reduce the startup delay, and the delay can be controlled by the client's playback rate. Additionally, through self-testing with the play demo, it was found that the delay had little impact.

`TRANS_BY_GPT3`

**Comment by xiaofeilong** [comment]
> 这个机制跟rtc的低延迟好像有冲突

**Comment by duiniuluantanqin** [comment]
I tested the functionality and there are no issues.

@xiaozhihong, please take a look and see if it is necessary to merge?

`TRANS_BY_GPT3`

**Comment by xiaofeilong** [comment]
Applicable scenarios include live streaming in a fast-paced environment.

`TRANS_BY_GPT3`

**Comment by quanqigu** [comment]
Experts, is this still appropriate? We are also facing the same issue. We need this fix.

`TRANS_BY_GPT3`

**Pull request closed without merging**